content-script.js

```diff
@@ -1,157 +1,254 @@
-const accessToken = 'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx'
-function rgbToHsl(r, g, b) {
-  r /= 255, g /= 255, b /= 255;
-
-  var max = Math.max(r, g, b), min = Math.min(r, g, b);
-  var h, s, l = (max + min) / 2;
-
-  if (max == min) {
-    h = s = 0; // achromatic
-  } else {
-    var d = max - min;
-    s = l > 0.5 ? d / (2 - max - min) : d / (max + min);
-
-    switch (max) {
-      case r: h = (g - b) / d + (g < b ? 6 : 0); break;
-      case g: h = (b - r) / d + 2; break;
-      case b: h = (r - g) / d + 4; break;
-    }
+/**
+ * @typedef {Object.<string, Array<{ id: number, notificationsItemDOM: Array<Element> }>>} NotificationInfo
+ */
 
-    h /= 6;
-  }
+/**
+ * @typedef {Object.<string, Array<{ id: number, labels: Array<{ color: string, description: string, name: string, url: string }> }>>} NotificationLabel
+ */
 
-  return [ h, s, l ];
-}
+const accessToken = 'xxxxxxxxxxxxxxxxxxxxxxxxxxxxx';
 
-const cache = new Map
-async function getIssueData(repo, id) {
-  const key = `${repo}#${id}`
-  if (cache.has(key)) {
-    return cache.get(key)
-  }
-  // TODO: remove this; all data should be fetched via GraphQL now.
-  const resp = await fetch(`https://api.github.com/repos/${repo}/issues/${id}`)
-  if (!resp.ok) {
-    cache.set(key, null)
-    return null
+/**
+ * query repo and id in /notifications page
+ *
+ * @returns {null | NotificationInfo}
+ */
+const notificationRepoAndID = () => {
+  const notificationsItems = document.querySelectorAll('li.notifications-list-item');
```

Comment:
looks like this lost the `:not(.watched)` selector, so we'll re-query all the time?

Reply:
We don't need `setInterval` / `MutationObserver` / `cache` and this.

**Reson:**
The github `/notification` page don't have `hot reload`. No changes are made to the front-end UI layer, even if a new notification comes in. Unless we do a page refresh.

Reply:
What about when selecting some notifications and clicking 'done'?

Reply:
oooh. You're right